packages/js/product-editor/src/contexts/validation-context/validation-provider.tsx

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ */
+import { createElement, useState } from '@wordpress/element';
+import { PropsWithChildren } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import {
+	ValidationErrors,
+	ValidationProviderProps,
+	Validator,
+	ValidatorResponse,
+} from './types';
+import { ValidationContext } from './validation-context';
+
+export function ValidationProvider< T >( {
+	initialValue,
+	children,
+}: PropsWithChildren< ValidationProviderProps< T > > ) {
+	const [ validators, setValidators ] = useState<
+		Record< string, Validator< T > >
+	>( {} );
```

Comment:
Yep at the beginning I thought validations to be exposed via context so anyone can use then in their own way. But yes, after having this other functions that gets access to validations I don't think we need to share them anyway. So yes, we can use `useRef` to avoid non necessary renders here. 
Thanks @louwie17 nice suggestion